synapse/cores/common.py

```diff
@@ -2619,6 +2622,109 @@ def _popCoreXact(self):
     def _getCoreXact(self, size):
         raise NoSuchImpl(name='_getCoreXact')
 
+    def _onAddSynTagDark(self, mesg):
+        tufo = mesg[1].get('tufo')  # type: tuple(str, dict)
+        tag = mesg[1].get('tag')  # type: str
+        self.addTufoDark(tufo, 'tag', tag)
+
+    def _onDelSynTagDark(self, mesg):
+        tufo = mesg[1].get('tufo')  #type: ((str, dict))
+        tag = mesg[1].get('tag')
+        self.delTufoDark(tufo, 'tag', tag)
+
```

Comment:
Amusingly, my original bits included these inline and I later moved them into being event driven.  I've left the subsequent syn:dark:* events in place.